Geometry/FindIntersections.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * @typedef {Object} Point
```

Comment:
This, and many of the other typedefs (Segment, Event, maybe Intersection) should probably be classes; the functions on them make sense as methods (for example a find intersection method for Segment).

If you decide to keep (some of) these as "implicit" "structs" for whatever reason, the `create...` functions are still useless boilerplate which should be removed IMO.